Sources/SwiftProtobufPluginLibrary/Descriptor.swift

```diff
@@ -372,7 +439,7 @@ public final class Descriptor {
     public var fullName: String { return containingType.fullName }
 
     /// The .proto file in which this ExtensionRange was defined.
-    public var file: FileDescriptor { return containingType.file }
+    public var file: FileDescriptor! { return containingType.file }
```

Comment:
This isn't possible :(

Reply:
```swift
if let file = descriptor.file {
  // do something with `file`
}
```
This will break if we change the type from `FileDescriptor!` to `FileDescriptor`: even though the former is implicitly unwrapped, it's still an `Optional` so it can be used like one everywhere (so switching over it would break too for example).

Reply:
This isn't a warning it just straight up generates an error

```swift
let nonOptional: Int = 1
let implicitOptiona: Int! = 1
if let i = nonOptional {} // error: Initializer for conditional binding must have Optional type, not 'Int'
if let i = implicitOptiona {}
```

I agree that this is annoying but it is still a breaking change and we never know if anybody out there does the `if let` dance.